File: Python/phases/nothrust_phase.py
import setup


def simulateNoThrustPhase(ypos_0, yvel_0, dt):
    time = 0
    ypos = ypos_0
    yvel = yvel_0
    m = setup.bottleconst["dry_mass"]
    
    rho_air = setup.physicsconst["air_density"]
    g = setup.physicsconst["gravity_acc"]
    
    dc = setup.bottleconst["drag_coeficient"]
    ab = setup.bottleconst["sec_area"]
    
    time_points = list()
    ypos_points = list()

    while (ypos > 0):
        drag_force = -1/2 * dc * rho_air * ab * yvel * abs(yvel)
        yvel += (drag_force / m - g) * dt
        ypos += yvel * dt
        time += dt
        
        time_points.append(time)
        ypos_points.append(ypos)
    
    return {
        "time": time_points,
        "ypos": ypos_points
    }

File: src/simulation/phases/nothrustphase.c
#include "simulation/simulate.h"


void SimulateNoThrustPhase(Simulation *simulation)
{
    double dt = simulation->dt;
    double time = 0;
    double y_acceleration = 0;
    double y_velocity = GetLastDoubleVector(&simulation->gas_phase_solution.y_velocity);
    double y_position = GetLastDoubleVector(&simulation->gas_phase_solution.y_position);
    double drag_force; 

    double g = simulation->physicsconst.gravity_acceleration;
    double rho_air = simulation->physicsconst.air_density;

    double mass = GetRocketVariable(&simulation->rocketvar, "dry_mass");
    double dc = GetRocketVariable(&simulation->rocketvar, "drag_coeficient");
    double ab = GetBottleSectionalArea(&simulation->rocketvar);

    while (y_position > 0)
    {
        drag_force = -1.0/2.0 * dc * rho_air * ab * y_velocity * fabs(y_velocity);
        y_velocity += (drag_force / mass - g) * dt;
        y_position += y_velocity * dt;
        time += dt;
        
        AppendDoubleVector(&simulation->nothrust_phase_solution.time, time);
        AppendDoubleVector(&simulation->nothrust_phase_solution.y_position, y_position);
        AppendDoubleVector(&simulation->nothrust_phase_solution.y_velocity, y_velocity);
    }
}

